src/scanner.cc

```diff
@@ -151,59 +155,184 @@ struct Scanner {
+            skip(lexer);
+          } else if (lexer->lookahead != '!') {
+            // Not a newline and not a comment, so not an
+            // end-of-statement
             return false;
+          }
         }
+
+        lexer->result_symbol = _END_OF_STATEMENT;
+        return true;
+    }
+
+    bool scan_start_line_continuation(TSLexer *lexer) {
+      // Now see if we should start a line continuation
+      in_line_continuation = (lexer->lookahead == '&');
+      if (!in_line_continuation) {
+        return false;
+      }
+      // Consume the '&'
+      advance(lexer);
+      lexer->result_symbol = _LINE_CONTINUATION;
+      return true;
+    }
+
+    bool scan_end_line_continuation(TSLexer *lexer) {
+      if (!in_line_continuation) {
+        return false;
+      }
+      // Everything except comments ends a line continuation
+      if (lexer->lookahead == '!') {
+        return false;
+      }
+
+      in_line_continuation = false;
+
+      // Consume any leading line continuation markers
+      if (lexer->lookahead == '&') {
         advance(lexer);
+      }
+      lexer->result_symbol = _LINE_CONTINUATION;
+      return true;
+    }
 
-        // Allow whitespace after line continuation
-        while (lexer->lookahead == ' ') {
-          skip(lexer);
-        }
+    bool scan_string_literal(TSLexer *lexer) {
+      const char opening_quote = lexer->lookahead;
 
-        // Allow comments after line continuation
-        if (lexer->lookahead == '!') return true;
+      if (opening_quote != '"' && opening_quote != '\'') {
+        return false;
+      }
 
-        // Consume end of line characters, we allow '\n', '\r\n' and
-        // '\r' to cover unix, MSDOS and old style Macintosh
-        if (lexer->lookahead == '\r') {
-          lexer->advance(lexer, false);
+      advance(lexer);
+      lexer->result_symbol = STRING_LITERAL;
+
+      while (lexer->lookahead != '\n') {
+        // Handle line continuations: MUST have both trailing '&' on
+        // first line AND leading '&' on second line. Various
+        // compilers do accept string literals missing the second '&'.
+        // Comments can also technically appear in the line
+        // continuation, but no thank you, we'll eat them as part of
+        // the string
+        if (lexer->lookahead == '&') {
+          advance(lexer);
+          // Consume blanks up to the end of the line or non-blank
+          while (std::iswblank(lexer->lookahead)) {
+            advance(lexer);
+          }
+          // If we hit the end of the line, consume all whitespace,
+          // including new lines
           if (lexer->lookahead == '\n') {
+            while (std::iswspace(lexer->lookahead)) {
               advance(lexer);
+            }
+            // If the next non-whitespace character isn't '&', this
+            // literal is technically ill-formed... but we could relax
+            // this to support compiler extensions
+            if (lexer->lookahead != '&') {
```

Comment:
Might be a good idea to relax this if some major modern compilers support it. Reading through the code I doesn't appear to be a critical token for disambiguating the token.